src/app/components/todo/todos.component.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from "react";
+import { ChangeEvent } from "react";
+import { DIContext } from "@helpers";
+import "./todo.style.css";
+
+const Todos: React.FC = () => {
+  const [name, setName] = React.useState("");
+
+  const inputValue = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+  };
+  const dependencies = React.useContext(DIContext);
+  const { todoService, translation } = dependencies;
+
+  const submitBtn = (e: React.FormEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    todoService.addTodo(name);
+    setName("");
```

Comment:
You should await for `todoService.addTodo(name);` before proceeding, right?

Imagine the implementation of TODOService changes and it makes a call to backend. Its not going to sync call. And setting name immediately after would be incorrect

Reply:
You should ensure that errors are handled

Consider where you have ~10MB (or some limit, I don't remember) of data in local storage. I believe local storage API should fail.

What's the experience looks like in that case?